spec/puppet/resource_api_spec.rb

```diff
@@ -408,105 +408,308 @@ def set(_context, changes)
     before(:each) do
       stub_const('Puppet::Provider::Canonicalizer', Module.new)
       stub_const('Puppet::Provider::Canonicalizer::Canonicalizer', provider_class)
+      Puppet.settings[:strict] = :off
     end
 
-    it { expect { described_class.register_type(definition) }.not_to raise_error }
+    context 'with get values that will not be canonicalized' do
```

Comment:
I'm having a hard time parsing this. Maybe "when `get` returns canonical values" ?